### DISCUSSION TITLE
Labels in Circuit Boxes

### DISCUSSION THREAD

**Discussion by gerakis** [FakeFishGames/Barotrauma#12920] [Ideas and suggestions]
I would like it if we could place written text inside of circuit boxes that explain how to use the circuit or how it works. Or even just a warning not to touch it. There is currently no way to do that.

It would also be nice if we could label the inputs and outputs.

**Comment by BlackMoons** [comment]
Absolutely. One of the big things about wiring my sub in components is I can put labels around the circuit (all hidden in game), since there is basically 0 chance im going to remember what it all does later without them.

**Comment by deltanedas** [comment]
you *can* do this in the sub editor by changing msg but it isnt visible ingame
documenting large circuits would help maintaining them a lot and when rewiring them ingame

**Reply by Saevon** [reply]
You can do this in game but you fuck with the interact msg field. Which I haven't found a way to reinsert the "[[Left Click]] Interact" with "[[Left Click]] Weapons Junction"  or "[[Left Click]] Docking Gate"

**Comment by FalcoGer** [comment]
I would also like to have labels available in circuit boxes, so that I may label a group of components or note down a specific formula that the circuits calculate for me.

It would be really helpful if one could give the inputs and outputs specific names. "SIGNAL_IN_1" isn't as descriptive as "Condition", or "SIGNAL_OUT_4" should be able to be renamed to "Fission Rate".

It'd be also nice if one could rename the entire circuit box, from "Circuit Box" to something more descriptive, such as "Door controller".

Additionally, I would like a line, box and/or polygon tool to put a delimiter around specific groups of components.

**Reply by Saevon** [reply]
see #4721 for the "Line around a specific group of components" as a nicer intuitive solution!  I also go over labelling inputs and outputs for the circuit box.
